packages/engine/src/renderer/systems/WebGLRendererSystem.ts

```diff
@@ -159,16 +182,76 @@ export class WebGLRendererSystem extends System {
       this.configurePostProcessing(entity);
     });
 
+    // count fps
+    let time = this.now();
+    this.frames++;
+    if (time > this.prevTime + this.interval) {
+      this.fps = Math.round((this.frames * this.interval) / (time - this.prevTime));
+      this.prevTime = time;
+      this.frames = 0;
+    }
+
+    this.changeQualityLevel(delta);    
+
     if(this.isInitialized)
       this.queryResults.renderers.all.forEach((entity: Entity) => {
         resize(entity);
-        getComponent<RendererComponent>(entity, RendererComponent).composer.render(delta);
+
+        if (this.qualityLevel >= 2) {
+          getComponent<RendererComponent>(entity, RendererComponent).composer.render(delta);
+          Engine.renderer.outputEncoding = LinearEncoding; // need this if postprocessing is used
+        }
+        else {
+          Engine.renderer?.render(Engine.scene, Engine.camera);
+          Engine.renderer.outputEncoding = sRGBEncoding; // need this if postprocessing is not used
```

Comment:
if in line 205 you are not sure that Engine.renderer is set, then add same optional chaining in line 206.
Engine.renderer?.outputEncoding = sRGBEncoding;

Reply:
unfortunately, it throws an error in line 206

Reply:
The left-hand side of an assignment expression may not be an optional property access.

Reply:
oh, sure, than
```js
if (Engine.renderer) {
          Engine.renderer.render(Engine.scene, Engine.camera);
          Engine.renderer.outputEncoding = sRGBEncoding; 
}
```

Reply:
Thanks. This is better solution to the problem